enumerables.rb

```diff
@@ -0,0 +1,156 @@
+# rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity,Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    Array(self).length.times do |idx|
+      yield(Array(self)[idx])
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    Array(self).length.times do |idx|
+      yield(Array(self)[idx], idx)
+    end
+    self
+  end
+
+  def my_select()
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    Array(self).my_each do |index_value|
+      next unless yield(index_value)
+
+      new_array.push(index_value)
+    end
+    new_array
+  end
+
+  def my_all?(argument = nil)
+    Array(self).my_each do |item|
+      if !block_given?
+        case argument
+        when argument.nil?
+          next if item
+        when argument.class == Class
+          next if item.is_a? argument
+        when argument.class == regexp
+          next if item =~ argument
+        when (item.is_a? Numeric) || (item.is_a? String)
+          next if item == argument
+        end
+      elsif yield(item)
+        next
+      end
+      return false
+    end
+    true
+  end
+
+  def my_any?(arr = nil)
+    Array(self).my_each do |arg|
+      if !block_given?
+        case arr
+        when arr.nil?
+          next unless arg
+        when arr.class == Class
+          next unless arg.is_a? arr
+        when arr.class == Regexp
+          next unless arg =~ arr
+        when (arg.is_a? Numeric) || (arg.is_a? String)
```

Comment:
- [ ] Same Problem of the my_all. You need to fix this for all the possible arguments.

![image](https://user-images.githubusercontent.com/61479648/90629436-c5345600-e1f5-11ea-88f2-3b2c10b74890.png)

Reply:
Issue fixed